packages/jest-runtime/src/index.js

```diff
@@ -318,25 +336,50 @@ class Runtime {
         id: modulePath,
         loaded: false,
       };
+
       moduleRegistry[modulePath] = localModule;
+
       if (path.extname(modulePath) === '.json') {
         localModule.exports = this._environment.global.JSON.parse(
           stripBOM(fs.readFileSync(modulePath, 'utf8')),
         );
       } else if (path.extname(modulePath) === '.node') {
         // $FlowFixMe
         localModule.exports = require(modulePath);
+      } else if (moduleName && isNativeModule) {
+        // If we come from a native module, or if we are a native module, then
+        // use native resolution. Native resolution differs from standard
+        // resolution on how requires are resolved (e.g. the "util" module
+        // requires "internal/util", which is not the NPM "internal" module,
+        // but something else).
```

Comment:
For this comment, how about:

Use a special resolution when requiring node's internal modules. For example the "util" module requires "internal/util" which isn't a physical directory.

It's shorter and conveys the same message.